drivers/sensor/nxp/qdec_tpm/qdec_nxp_tpm.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_tpm_qdec
+
+#include <errno.h>
+#include <soc.h>
+#include <fsl_tpm.h>
+
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor/qdec_mcux.h>
+#include <zephyr/logging/log.h>
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+LOG_MODULE_REGISTER(qdec_tpm, CONFIG_PWM_LOG_LEVEL);
+
+struct qdec_tpm_config {
+	TPM_Type *base;
+	tpm_clock_prescale_t prescale;
+	const struct pinctrl_dev_config *pincfg;
+	const tpm_phase_params_t phaseParams;
+};
+
+struct qdec_tpm_data {
+	int32_t count;
+	uint16_t counts_per_revolution;
+};
+
+static int qdec_tpm_attr_set(const struct device *dev, enum sensor_channel ch,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	struct qdec_tpm_data *data = dev->data;
+
+	if (ch != SENSOR_CHAN_ROTATION) {
+		return -ENOTSUP;
+	}
+
+	switch ((enum sensor_attribute_qdec_mcux)attr) {
+	case SENSOR_ATTR_QDEC_MOD_VAL:
+		if (!IN_RANGE(val->val1, 1, UINT16_MAX)) {
+			LOG_ERR("SENSOR_ATTR_QDEC_MOD_VAL value invalid");
+			return -EINVAL;
+		}
+		data->counts_per_revolution = val->val1;
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int qdec_tpm_attr_get(const struct device *dev, enum sensor_channel ch,
+			     enum sensor_attribute attr, struct sensor_value *val)
+{
+	struct qdec_tpm_data *data = dev->data;
+
+	if (ch != SENSOR_CHAN_ROTATION) {
+		return -ENOTSUP;
+	}
+
+	switch ((enum sensor_attribute_qdec_mcux)attr) {
```

Comment:
Shouldn't be using private attributes from another driver. Maybe time to introduce a new common attribute

Reply:
Accidentally migrated from the other QDEC driver, created it's own definition now.